components/board/pi/impl/board.go

```diff
@@ -679,36 +679,36 @@ func (pi *piPigpio) AnalogByName(name string) (board.Analog, error) {
 // NOTE: During board setup, if a digital interrupt has not been created
 // for a pin, then this function will attempt to create one with the pin
 // number as the name.
-func (pi *piPigpio) DigitalInterruptByName(name string) (board.DigitalInterrupt, bool) {
+func (pi *piPigpio) DigitalInterruptByName(name string) (board.DigitalInterrupt, error) {
 	pi.mu.Lock()
 	defer pi.mu.Unlock()
 	d, ok := pi.interrupts[name]
 	if !ok {
 		var err error
 		if bcom, have := broadcomPinFromHardwareLabel(name); have {
 			if d, ok := pi.interruptsHW[bcom]; ok {
-				return d, ok
+				return d, nil
 			}
 			d, err = CreateDigitalInterrupt(DigitalInterruptConfig{
 				Name: name,
 				Pin:  name,
 				Type: "basic",
 			})
 			if err != nil {
-				return nil, false
+				return nil, err
 			}
 			if result := C.setupInterrupt(C.int(bcom)); result != 0 {
 				err := picommon.ConvertErrorCodeToMessage(int(result), "error")
-				pi.logger.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
-				return nil, false
+				err1 := errors.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
```

Comment:
You can join the error statements, I think you can use fmt for the one you're making and invoke it directly in the input to errors.Join()
https://pkg.go.dev/errors#Join

Reply:
```suggestion
				return nil, errors.Errorf("Unable to set up interrupt on pin %s: %s", name, err)
```

Reply:
what about just returning this directly? not sure why I didn't do that

Reply:
great